subsys/net/l2/canbus/canbus.c

```diff
@@ -175,6 +192,23 @@ static bool dest_is_translator(struct net_pkt *pkt)
 		get_dest_lladdr(pkt) == NET_CAN_ETH_TRANSLATOR_ADDR);
 }
 
+#ifdef CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR
+static size_t total_lladdr_len(struct net_pkt *pkt)
+{
+	/* This pkt will be farowarded to Ethernet
+	 * Destination MAC is carried inline, source is going to be extended
+	 */
+	if (dest_is_translator(pkt)) {
+		return sizeof(struct net_eth_addr) +
+		       sizeof(struct net_can_lladdr);
+	}
+#else
```

Comment:
This looks ugly, can't we use same function prototype for total_lladdr_len()?
So something like:
```
static size_t total_lladdr_len(struct net_pkt *pkt)
{
#if defined(CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR)
	/* This pkt will be farowarded to Ethernet
	 * Destination MAC is carried inline, source is going to be extended
	 */
	if (dest_is_translator(pkt)) {
		return sizeof(struct net_eth_addr) +
		       sizeof(struct net_can_lladdr);
	}
#else 
	ARG_UNUSED(pkt);
#endif 

	return 2U * sizeof(struct net_can_lladdr);
}
```

Reply:
Sure